components/gitpod-cli/cmd/ports-visibility.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	gitpod "github.com/gitpod-io/gitpod/gitpod-cli/pkg/gitpod"
+	serverapi "github.com/gitpod-io/gitpod/gitpod-protocol"
+	"github.com/spf13/cobra"
+)
+
+// portsVisibilityCmd change visibility of port
+var portsVisibilityCmd = &cobra.Command{
+	Use:   "visibility <port>[:private][:public]",
+	Short: "Make a port public or private",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		portVisibility := args[0]
+		s := strings.Split(portVisibility, ":")
+		if len(s) != 2 {
+			log.Fatal("cannot parse args, should be something like `3000:public` or `3000:private`")
+		}
+		port, err := strconv.Atoi(s[0])
+		if err != nil {
+			log.Fatal("port should be integer")
+		}
+		visibility := s[1]
+		if visibility != serverapi.PortVisibilityPublic && visibility != serverapi.PortVisibilityPrivate {
+			log.Fatalf("visibility should be `%s` or `%s`", serverapi.PortVisibilityPublic, serverapi.PortVisibilityPrivate)
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
```

Comment:
15 seconds is a longer timeout than all other `gp` commands. Does it need to be this long?

Reply:
Yep, I create this PR to test `openPort` so I increased this timeout duration 😂. Will reduce it to 5s